viewer/packages/data-providers/src/utilities/utils.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import { TableExpressionEqualsFilterV3 } from '@cognite/sdk/dist/src';
+import { PointCloudObject, PointCloudObjectMetadata } from '../pointcloud-stylable-object-providers/types';
+import { StylableObject } from '../pointcloud-stylable-object-providers/StylableObject';
+import { ClassicDataSourceType, DMDataSourceType, DataSourceType } from '../DataSourceType';
+
+/**
+ * Type guard to check if a point cloud object contains data type DMPointCloudDataType
+ * @param pointCloudObject - The object to check
+ * @returns True if the object is of type `PointCloudObject<DMPointCloudDataType>`, false otherwise
+ */
+export function isDMPointCloudDataTypeObject(
+  pointCloudObject: PointCloudObject<DataSourceType>
+): pointCloudObject is PointCloudObject<DMDataSourceType> {
+  return isDMPointCloudDataType(pointCloudObject) && hasStylableObject(pointCloudObject);
+}
+
+/**
+ * Type guard to check if a point cloud object contains data type ClassicPointCloudDataType
+ * @param pointCloudObject - The object to check
+ * @returns True if the object is of type `PointCloudObject<ClassicPointCloudDataType>`, false otherwise
+ */
+export function isClassicPointCloudDataTypeObject(
+  pointCloudObject: PointCloudObject<DataSourceType>
+): pointCloudObject is PointCloudObject<ClassicDataSourceType> {
+  return isClassicPointCloudDataType(pointCloudObject) && hasStylableObject(pointCloudObject);
+}
```

Comment:
Are these two necessary? (Since we have `isDMPointCloudDtaaType` etc. below?)

Reply:
Also, I see this file mentions `PointCloudDataType` a lot - it was an old name, so maybe we can update to e.g. `DataSourceType`?